pkg/log/log.go

```diff
@@ -236,3 +239,20 @@ func entryToJSON(entry *logrus.Entry) []byte {
 
 	return bytes
 }
+
+// SafeDumpPodObject redacts commands and args in Pod manifest to hide sensitive info,
+// converts Pod object into string and returns it
+func SafeDumpPodObject(pod *v1.Pod) string {
```

Comment:
This function modifies `pod`. Either:
- it should be noted in the function documentation; or
- the function should create a deep copy.

This is not an issue in the current usage of this function. But, if the function were to be reused elsewhere, this is surprising behavior and it may actually introduce software defects (a.k.a bugs).

Reply:
That makes sense. Thanks a lot for letting me know. I will keep that in mind. Also, will fix this in the follow-up